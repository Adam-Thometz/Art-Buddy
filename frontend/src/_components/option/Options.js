import React from "react";

import './Options.css';

const Options = ({ children, width = '100%' }) => {
  return (
    <div className="Options" style={{width}}>
      {children}
    </div>
  );
};

export default Options;