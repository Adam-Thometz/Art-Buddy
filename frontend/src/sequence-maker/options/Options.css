.Options {
  display: flex;
  justify-content: center;
  margin: 0 auto;
}

.Options-option {
  margin: 10px;
}

.Options-img {
  height: 200px;
  width: 200px;
}

.Options-description {
  font-weight: bold;
}