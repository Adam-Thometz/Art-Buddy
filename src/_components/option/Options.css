.Options {
  display: flex;
  justify-content: space-evenly;
  margin: 0 auto;
}