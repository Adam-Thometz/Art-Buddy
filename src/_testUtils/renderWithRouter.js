import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;