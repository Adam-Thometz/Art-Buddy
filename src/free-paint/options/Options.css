.Options {
  display: flex;
  flex-direction: column;
}

.Options-shape-button {
  width: 45px;
  height: 45px;
}