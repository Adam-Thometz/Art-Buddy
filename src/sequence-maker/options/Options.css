.Options {
  display: flex;
  margin: 0 auto;
  padding: 20px;
}

.Options-img {
  height: 200px;
  width: 200px;
}

.Options-description {
  font-weight: bold;
}