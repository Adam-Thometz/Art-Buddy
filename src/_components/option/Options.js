import './Options.css';

const Options = ({ children, width = '100%' }) => {
  return (
    <section className="Options" style={{width}}>
      {children}
    </section>
  );
};

export default Options;