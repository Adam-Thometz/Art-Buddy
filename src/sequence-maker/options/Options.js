import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { addToSequence } from "../../actions";
import soundInfo from "../soundInfo";
import './Options.css';

const Options = () => {
  const dispatch = useDispatch()
  const category = useSelector(state => state.sequenceMaker.category);
  const sequence = useSelector(state => state.sequenceMaker.sequence);
  const options = soundInfo[category];

  const add = e => {
    if (sequence.some(block => !block)) dispatch(addToSequence(e.target.classList[1]));
  }

  return (
    <div className="Options">
      {Object.keys(options).map(option => (
        <div className="Options-option">
          <img className={`Options-img ${option}`} src={options[option].image} alt={options[option].alt} onClick={add} />
          <p className="Options-description">{options[option].alt}</p>
        </div>
      ))}
    </div>
  );
};

export default Options;