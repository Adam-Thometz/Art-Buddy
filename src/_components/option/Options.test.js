import React from "react";
import renderWithRouter from "../../_testUtils/renderWithRouter";
// import { screen } from "@testing-library/react";
import Options from "./Options";

describe('Options component', () => {
  it('renders without crashing', () => {
    renderWithRouter(<Options />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithRouter(<Options />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('matches snapshot of smaller window', () => {
    const { asFragment } = renderWithRouter(<Options width="65%" />);
    expect(asFragment()).toMatchSnapshot();
  });
})