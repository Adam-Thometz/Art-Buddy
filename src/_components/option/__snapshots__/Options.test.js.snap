// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Options"
    style="width: 100%;"
  />
</DocumentFragment>
`;

exports[`Options component matches snapshot of smaller window 1`] = `
<DocumentFragment>
  <div
    class="Options"
    style="width: 65%;"
  />
</DocumentFragment>
`;
